src/dawn/Optimizer/PassTemporaryToStencilFunction.cpp

```diff
@@ -294,13 +285,16 @@ bool PassTemporaryToStencilFunction::run(
     const std::shared_ptr<StencilInstantiation>& stencilInstantiation) {
 
   OptimizerContext* context = stencilInstantiation->getOptimizerContext();
+
+  if(!context->getOptions().PassTmpToFunction)
+    return true;
+
   DAWN_ASSERT(context);
```

Comment:
should this assert be above the check of the options to ensure that we do not segfault?